### PR TITLE
Move ActionMailer configuration to initializer

### PR DESCRIPTION
**Pull request by dsander**
Prevents encoding issues in the generated YAML file when using especial characters like !

#1584

### CODE REVIEW

**Comment by dsander** [comment]
Moving the database config to an initializes does not seem to be possible, we could use the `DATABASE_URL` environment variable, but I don't know if that would work with unix sockets.

**Comment by cantino** [comment]
Nice @dsander. This looks right to me.

**Comment by albertsun** [comment]
I'll point out that some (albeit outdated) deploy systems like Capistrano depend on creating `database.yml` as a symlink or a file that's generated at deploy time. Our old deploy system used to do the same for `smtp.yml`. We now use environment variables but some people with older Huginn deploys might have trouble with this.

Might want to make a more prominent changelog note or something

**Comment by cantino** [comment]
Good point.  I still use Capistrano and it may try to manage database.yml.  Maybe best to leave it as-is for now?

**Comment by dsander** [comment]
Oh yes, that is very true. I think our current capistrano deployment recipe would not be affected, but I agree that it is a very common thing to do.
It looks like that we do not mention modifying the directly `smtp.yml` anywhere, so I don't think it needs a prominent place. I might find some time over the weekend to update the changelog with the recent changes and add a note there.

**Comment by cantino** [comment]
Sounds good.  This looks good to me.

File: config/initializers/action_mailer.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
-# Read smtp config out of a config/smtp.yml file
-
-smtp_config = YAML::load(ERB.new(File.read(Rails.root.join('config', 'smtp.yml'))).result)
-if smtp_config.keys.include? Rails.env
-  ActionMailer::Base.smtp_settings = smtp_config[Rails.env].symbolize_keys
-end
\ No newline at end of file
+ActionMailer::Base.smtp_settings = {
+  address: ENV['SMTP_SERVER'] || "smtp.gmail.com",
+  port: ENV['SMTP_PORT'] || 587,
```
Comment (knu) [inline config/initializers/action_mailer.rb]: No conversion to integer needed?

File: config/initializers/action_mailer.rb [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+  domain: ENV['SMTP_DOMAIN'],
+  authentication: ENV['SMTP_AUTHENTICATION'] || "plain",
+  enable_starttls_auto: ENV['SMTP_ENABLE_STARTTLS_AUTO'] == 'true' ? true : false,
```
Comment (knu) [inline config/initializers/action_mailer.rb]: Maybe `? true : false` can be omitted.

File: config/initializers/action_mailer.rb (hunk 1 above)
Comment (dsander) [inline config/initializers/action_mailer.rb]: I don't think so, previously we dumped the ENV variable (which always is a string) into the YAML file, `ActionMailer` probably handles the type casting.

File: config/initializers/action_mailer.rb (hunk 2 above)
Comment (dsander) [inline config/initializers/action_mailer.rb]: True :)

**Comment by dsander** [comment]
Thanks! I updated the changelog in #1602

**Pull request merged**